benches/primitives.rs

```diff
@@ -0,0 +1,68 @@
+use std::array;
+
+use criterion::{criterion_group, criterion_main, BenchmarkId, Criterion};
+use ff::Field;
+use orchard::primitives::{
+    poseidon::{self, ConstantLength, OrchardNullifier},
+    sinsemilla,
+};
+
+use pasta_curves::pallas;
+#[cfg(unix)]
+use pprof::criterion::{Output, PProfProfiler};
+use rand::{rngs::OsRng, Rng};
+
+fn bench_primitives(c: &mut Criterion) {
+    let mut rng = OsRng;
+
+    {
+        let mut group = c.benchmark_group("Poseidon");
+
+        let message = [pallas::Base::random(rng), pallas::Base::random(rng)];
+
+        group.bench_function("2-to-1", |b| {
+            b.iter(|| poseidon::Hash::init(OrchardNullifier, ConstantLength).hash(message))
+        });
+    }
+
+    {
+        let mut group = c.benchmark_group("Sinsemilla");
+
+        let hasher = sinsemilla::HashDomain::new("hasher");
+        let committer = sinsemilla::CommitDomain::new("committer");
+        let bits: Vec<bool> = (0..1086).map(|_| rng.gen()).collect();
+        let r = pallas::Scalar::random(rng);
+
+        // Benchmark the input sizes we use in Orchard:
+        // - 510 bits for Commit^ivk
+        // - 520 bits for MerkleCRH
```

Comment:
We technically don't commit `MerkleCRH`, but I don't think that really matters here.

Reply:
I decided to include it in the benchmarks because `520` bits of input corresponds to a 512-bit input with padding, which is something other people might want benchmarks for.